### ISSUE TITLE
Unable to start Prometheus when Knative enabled. 

### ISSUE THREAD

**Issue by weimeilin79** [apache/camel-k#1459]
Try enable prometheus for a knative service, but had error.. 

```
kamel run API.java --open-api openapi.yaml --trait prometheus.enabled=true
```

```
Cannot reconcile Integration api: error executing post actions: error during replace resource: could not create or replace resource api: admission webhook "validation.webhook.serving.knative.dev" denied the request: validation failed: Port name prometheus is not allowed: spec.template.spec.containers[0].ports
Name must be empty, or one of: 'h2c', 'http1'
```

**Comment by davsclaus** [comment]
@nicolaferraro I think this would be good to get done before GA

**Comment by nicolaferraro** [comment]
This is for sure a cool feature, but we need to investigate if it's currently technically possible without changes in Knative, because afaik Knative itself publishes a prometheus endpoint and exposes metrics for their internal prometheus and for the pod autoscaler.

**Comment by astefanutti** [comment]
It should be fixed with #1478. I've tested it with Knative auto-scaling:

<img width="1277" alt="Screenshot 2020-06-03 at 18 08 17" src="https://user-images.githubusercontent.com/366207/83661463-77c92500-a5c6-11ea-8ba2-ed4d8f945cb7.png">

It shows discontinuities when the integration scales to zero. The metrics are reset per pods, which I think is acceptable.
